music_assistant/providers/opensubsonic/sonic_provider.py

```diff
@@ -365,9 +366,16 @@ async def get_library_tracks(self) -> AsyncGenerator[Track, None]:
 
     async def get_album(self, prov_album_id: str) -> Album:
         """Return the requested Album."""
+        sonic_album: SonicAlbum
+        sonic_info = None
         try:
-            sonic_album: SonicAlbum = await self._run_async(self.conn.get_album, prov_album_id)
-            sonic_info = await self._run_async(self.conn.get_album_info2, aid=prov_album_id)
+            sonic_album = await self._run_async(self.conn.get_album, prov_album_id)
+            try:
+                sonic_info = await self._run_async(self.conn.get_album_info2, aid=prov_album_id)
+            except HTTPError as e:
```

Comment:
This is the wrong place to handle a 404. If the LastFM API key is not configured then your Open Subsonic server should be returning a response with the code for DataNotFoundError. The provider code would then need to place the call `to get_album_info2` in its own try block so that we can proceed with the basic info if that is all the server has.